### DISCUSSION TITLE
Identifying a client on an Authentication SPI

### DISCUSSION THREAD

**Discussion by oculos** [keycloak/keycloak#24539] [Q&A]
Hi,

Is it possible to find out which client the user is coming from when authenticating? 
I imagine that setting up oidc claims for each client would do the trick, but is there a more straightforward way to get this info on an Authenticator code? 

Best,
Francis

**Comment by danielFesenmeyer** [comment]
I think what you are searching for is: `AuthenticationFlowContext.getAuthenticationSession().getClient()`

**Reply by oculos** [reply]
Perfect, that did the trick!  Thank you!
